stock_picking_line_sequence/models/stock.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#    Copyright 2015 Camptocamp SA
+# © 2015 Eficent Business and IT Consulting Services S.L.
+# © 2015 Serpent Consulting Services Pvt. Ltd.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+
+from openerp import models, api, fields
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+    _order = 'date_expected desc, sequence, id'
+
+    sequence = fields.Integer()
+    sequence2 = fields.Integer(help="Shows the sequence in the Stock Move.",
+                               related='sequence', readonly=True, default=9999)
+
+    @api.model
+    def create(self, values):
+        move = super(StockMove, self).create(values)
+        # We do not reset the sequence if we are copying a complete stock move
+        if 'keep_line_sequence' not in self.env.context:
+            move.picking_id._reset_sequence()
+        return move
+
+    @api.one
+    def copy(self, default=None):
+        if not default:
+            default = {}
+        if 'keep_line_sequence' not in self.env.context:
+            default['sequence'] = 9999
+        return super(StockMove, self).copy(default)
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.depends('move_lines')
+    def _compute_max_line_sequence(self):
+        for picking in self:
+            picking.max_line_sequence = (
+                max(picking.mapped('move_lines.sequence') or [0]) + 1
+                )
+
+    max_line_sequence = fields.Integer(string='Max sequence in lines',
+                                       compute='_compute_max_line_sequence')
+
+    @api.multi
+    def _reset_sequence(self):
+        for rec in self:
+            current_sequence = 1
+            for line in rec.move_lines:
+                line.write({'sequence': current_sequence})
+                current_sequence += 1
+
+    @api.multi
+    def write(self, values):
+        res = super(StockPicking, self).write(values)
+        for rec in self:
+            rec._reset_sequence()
```

Comment:
IMO this makes no sense, the sequence is required to be kept to compare when invoicing or with the sale order (related lines will have same sequence number). If I change a minor thing in the picking the sequence is reset and the relation is lost.